### PR TITLE
Add physics and chemistry constants + translation

### PR DESCRIPTION
**Pull request by ifndev**
(suggested by @adriweb and @tjhorner , see #117)

Implementation of physics and chemistry constants in the toolbox
It can be useful to avoid errors when copying them from the subject during an exam or a test.

It adds the following approximative values (based on the ones given in the French baccalaureate and exercise books):

_(physics)_
Speed of light
Gravitational Constant
Coulomb's Constant
Elementary charge
Mass of a nucleon
Mass of a proton
Mass of a neutron
Mass of an electron
Plank constant

_(chemistry)_
Avogadro constant

I decided to include both the general mass of a nucleon and the mass of a proton/neutron because i'm unsure about those given at the exams in different countries, it may be confusing tough.

It may be necessary to check translations

**EDIT**: I removed most of the specials characters in translations, but i figured out they were replaced automatically... I won't change anything for the moment, up to you to decide whether or not they should be deleted

### CODE REVIEW

**Comment by 0b101** [comment]
Here are some screenshots. I would recommend that you move the constants node lower.
C1C2
C3C4

**Comment by boricj** [comment]
I would expect constants given by the calculator to be far more precise than that.

**Comment by Lesquivemeau** [comment]
Well, i partly agree with you but unfortunately the values given at the bachalaureate are these ones

(after checking, it seems that the nucleon's approximate mass is the one used most of the time)

**Comment by adriweb** [comment]
> Well, i partly agree with you but unfortunately the values given at the baccalaureate are these ones

I'm not sure how that's relevant though - if the user is given specific numbers to use he will input them directly. But if he wants to look them up and use them independently, he should be able to do so directly from the calc with greater precision.

From what I see on the Nspire CX CAS, constants are given with 10 sig. figs. (and with units built-in, which allows conversions)

**Comment by Lesquivemeau** [comment]
Interesting idea, giving the choice to the user would be a great idea.
Do you think it would be more relevant in the calculator settings (in order not to bloat the toolbox) or in the toolbox ?

**Comment by adriweb** [comment]
Ah, no, I wasn't clear enough apparently ^^

In my opinion, the calc should only provide greater-precision values because I think that users needing to use specific ones (given by exams or for any other reasons) would type them directly and not bother fetching them from the menus.

**Comment by Lesquivemeau** [comment]
Well, it's a good point, but when in an exam copying the values from the subject can result in several errors. However, i don't think having a complete app dedicated to constants (like on the 83pce) would be a good idea, as it would be unintuitive to use

**Comment by Lesquivemeau** [comment]
I will change actual values for more precise ones, but i strongly consider adding a menu to chose between precise and approximate values

File: apps/shared.universal.i18n [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-c_G = "6.67*10^-11"
-c_k = "9.0*10^9"
-c_e = "1.60*10^-19"
-c_Na = "6.02*10^23"
-c_mg = "1.67*10^-27"
-c_mp = "1.673*10^-27"
-c_mn = "1.675*10^-27"
-c_me = "9.109*10^-31"
-c_h = "6.63*10^-34"
+c_c = "299 792 458"
```
Comment (0b101) [inline apps/shared.universal.i18n]: "299 792 458" should be written as "299792458"
```suggestion
c_c = "299792458"
```

**Comment by nsajko** [comment]
Adding these and other constants is a must for NumWorks because other calculators have them, even much cheaper ones; but, @Lesquivemeau, you can not just pick arbitrary "approximations" to well known constants. Every constant definition should be as close to the actual (defined or known) constant as possible, and for maintainability each constant definition should be accompanied by a reference (preferably URI), like here: https://github.com/golang/go/blob/54abb5ff868bf20239bd7960038a28c3f31ba25e/src/math/const.go#L10 .

For example, NIST and CODATA ( https://physics.nist.gov/cuu/Constants/index.html https://www.codata.org/committees-and-groups/fundamental-physical-constants ) have a different opinion of Avogadro's constant's definition than you ;) .

One thing that should be noted is that it is wrong to just copy constant definitions from some textbook or exam because there are many different textbooks and exams over the world.

**Comment by sagessylu** [comment]
@ifndev There is a typo in `apps/toolbox.fr.i18n` :
`c_hDesc = "Constante de Plank"` should be replaced with
`c_hDesc = "Constante de Planck"`

**Comment by RedGl0w** [comment]
>  @ifndev There is a typo in `apps/toolbox.fr.i18n` :
> `c_hDesc = "Constante de Plank"` should be replaced with
> `c_hDesc = "Constante de Planck"`

You can also use suggestion in pr for this (click on the line you want to change, click on the + on le left, and then there's a button in the top bar for adding suggestion)

File: apps/toolbox.fr.i18n [hunk 2]
```diff
@@ -53,3 +56,12 @@ RandomAndApproximation = "Aléatoire et approximation"
...
 PrimeFactorDecomposition = "Décomposition en facteurs premiers"
+c_cDesc = "Vitesse de la lumiere"
+c_GDesc = "Constante de gravitation"
+c_kDesc = "Constante de Coulomb"
+c_eDesc = "Charge élémentaire"
+c_NaDesc = "Nombre d'Avogadro"
+c_mpDesc = "Masse d'un proton"
+c_mnDesc = "Masse d'un neutron"
+c_meDesc = "Masse d'un électron"
+c_hDesc = "Constante de Plank"
```
Comment (sagessylu) [inline apps/toolbox.fr.i18n:67]: There is a typo : it should be `c_hDesc = "Constante de Planck"`

File: apps/toolbox.fr.i18n [hunk 3]
```diff
@@ -53,3 +56,12 @@ RandomAndApproximation = "Aléatoire et approximation"
 RandomFloat = "Nombre décimal dans [0,1["
 RandomInteger = "Entier aléatoire dans [a,b]"
 PrimeFactorDecomposition = "Décomposition en facteurs premiers"
+c_cDesc = "Vitesse de la lumiere"
```
Comment (sagessylu) [inline apps/toolbox.fr.i18n:59]: There is a typo : it should be `c_cDesc = "Vitesse de la lumière"`

**Pull request closed without merging**